tuf/data/roles.go

```diff
@@ -101,6 +113,111 @@ type Role struct {
 	Email            string   `json:"email,omitempty"`
 }
 
+// BaseRole is an internal representation of a root/targets/snapshot/timestamp role, with its public keys included
+type BaseRole struct {
+	Keys      map[string]PublicKey `json:"keys"`
+	Name      string               `json:"name"`
+	Threshold int                  `json:"threshold"`
+}
+
+// ListKeys retrieves the public keys valid for this role
+func (b BaseRole) ListKeys() KeyList {
+	return listKeys(b.Keys)
+}
+
+// ListKeyIDs retrieves the list of key IDs valid for this role
+func (b BaseRole) ListKeyIDs() []string {
+	return listKeyIDs(b.Keys)
+}
+
+// DelegationRole is an internal representation of a delegation role, with its public keys included
+type DelegationRole struct {
+	BaseRole
+	Paths            []string `json:"paths,omitempty"`
+	PathHashPrefixes []string `json:"path_hash_prefixes,omitempty"`
+}
+
+// ListKeys retrieves the public keys valid for this role
+func (d DelegationRole) ListKeys() KeyList {
```

Comment:
I'm think these function definitions necessary?   Since `DelegationRole` embeds `BaseRole`, calling `d.ListKeys()` [should invoke the embedded `BaseRole`'s `ListKeys()`](https://golang.org/doc/effective_go.html#embedding), which is the correct behavior for `DelegationRole` since the keys are on `BaseRole` anyway.